### PR TITLE
Remove almost all uses of db::item_type

### PR DESCRIPTION
**Pull request by kidder**
## Proposed changes

`db::item_type<Tag>` is replaced with either the known type or `typename Tag::type`

To help with review and rebasing I separated into separate commits by directory.  I plan to squash into a single commit after review.

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [x] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/IO/Importers/VolumeDataReader.hpp [hunk 1]
```diff
@@ -67,7 +67,8 @@ struct InitializeVolumeDataReader {
     return std::make_tuple(
```
Comment (nilsdeppe) [inline src/IO/Importers/VolumeDataReader.hpp]: This is likely going to cause a nightmare of conflicts with my PR... Could we drop it and I remove them either in my PR or in a subsequent one? I'm changing the types and number of I think literally every single one you've touched...

**Review by nilsdeppe — COMMENTED** [review]
Looks good, one request that would make my life a lot easier. Let me know if you disagree or it's not feasible :)

File: src/IO/Importers/VolumeDataReader.hpp (hunk 1 above)
Comment (kidder) [reply src/IO/Importers/VolumeDataReader.hpp]: no problem, just the IO commit?

**Review by nilsdeppe — DISMISSED** [review]

**Comment by nilsdeppe** [comment]
There are a couple correct clang-tidy things, but I'm fine ignoring since it's just collateral
```
2020-08-22T06:10:21.2744392Z Suppressed 35109 warnings (34878 in non-user code, 231 NOLINT).
2020-08-22T06:10:21.2744840Z /__w/spectre/spectre/src/DataStructures/Tensor/EagerMath/Norms.hpp:24:8: error: 'auto tensor_element' can be declared as 'const auto *tensor_element' [llvm-qualified-auto,-warnings-as-errors]
2020-08-22T06:10:21.2745194Z   for (auto tensor_element = tensor.begin(); tensor_element != tensor.end();
2020-08-22T06:10:21.2745339Z        ^
2020-08-22T06:10:21.2745437Z note: this fix will not be applied because it overlaps with another fix
2020-08-22T06:10:21.2745932Z /__w/spectre/spectre/src/DataStructures/Tensor/EagerMath/Norms.hpp:24:8: error: 'auto tensor_element' can be declared as 'const auto *tensor_element' [readability-qualified-auto,-warnings-as-errors]
2020-08-22T06:10:21.2746163Z note: this fix will not be applied because it overlaps with another fix
2020-08-22T06:10:21.2746767Z Use -header-filter=.* to display errors from all non-system headers. Use -system-headers to display errors from system headers as well.
```

**Comment by kidder** [comment]
When I added the `const auto*` it then complained about doing pointer arithmetic on `tensor_element`

**Comment by nilsdeppe** [comment]
Ah okay, let's leave it then...

**Comment by nilsdeppe** [comment]
@wthrowe would you be interested in and have time to review this? If not, we can find someone else :) Thanks!

**Comment by wthrowe** [comment]
Looks fine.  Squash.

**Comment by kidder** [comment]
@wthrowe @nilsdeppe squashed; thanks!

**Review by nilsdeppe — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**